### PR TITLE
add aws-cdi-sdk 2.2.0

### PR DESCRIPTION
**Pull request by dvirtz**
Specify library name and version:  **aws-cdi-sdk/2.2.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by dvirtz** [comment]
@jgsogo @SSE4 @madebr 
This package requires a non-default option (`monitoring`) of `aws-sdk-cpp` and thus fails to find a prebuilt binary.
Should I open a PR to change that default value to match this package?

**Comment by SSE4** [comment]
> @jgsogo @SSE4 @madebr
> This package requires a non-default option (`monitoring`) of `aws-sdk-cpp` and thus fails to find a prebuilt binary.
> Should I open a PR to change that default value to match this package?

I think so, if there is no other way, let's try to change the defaults of `aws-sdk-cpp`

**Comment by dvirtz** [comment]
@SSE4 this is not blocked any more

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+    _cmake = None
+
+    def configure(self):
+        self.options["aws-sdk-cpp"].monitoring = True
+        self.options["aws-sdk-cpp"].shared = True
+        self.options["aws-libfabric"].shared = True
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:30]: These can be overridden by consumers so if these are requirements then they should be moved to a validate method

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return self._cmake
+
+    def build(self):        
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        # build aws-cpp-sdk-cdi
+        cmake = self._configure_cmake()
+        cmake.build()
+        cmake.install()
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:66]: This should be called in the package method

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        autotools = self._configure_autotools()
+        with tools.chdir(self._source_subfolder):
+            vars = autotools.vars
+            # configure autotools to find aws-cpp-sdk-cdi
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:71]: Can you please explain why this is required? I do not understand why it's being build twice 😕

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    def configure(self):
+        self.options["aws-sdk-cpp"].monitoring = True
+        self.options["aws-sdk-cpp"].shared = True
+        self.options["aws-libfabric"].shared = True
+
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("The aws-cdi-sdk con only be built on Linux.")
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:35]: Windows could be supported. https://github.com/aws/aws-cdi-sdk/blob/mainline/INSTALL_GUIDE_WINDOWS.md#build-the-aws-cdi-sdk

```suggestion
            raise ConanInvalidConfiguration("This recipe currently only supports Linux. Feel free to contribute other platforms!")
```

File: recipes/aws-cdi-sdk/all/patches/001-Makefile.patch [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+-all : libfabric libsdk test docs docs_api $(EXTRA_ALL_TARGETS)
++all : libsdk test docs docs_api $(EXTRA_ALL_TARGETS)
+ 
+ # Build only the libraries
+ .PHONY : lib
+-lib : libfabric libsdk
+-
+-# Ensure that the location of the AWS SDK source code tree was specified unless explicitly opted out. Define and augment
+-# some variables needed for building and linking to the AWS SDK.
+-ifeq ($(require_aws_sdk),yes)
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/patches/001-Makefile.patch:118]: This does not seem required to build the library 🤔 can you explain what this patch is for please?

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        return self._autotools
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CPP_STANDARD"] = self._cmake.definitions.get("CONAN_CMAKE_CXX_STANDARD", "11")
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:55]: I would have expected this to come from the users profile/settings

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
Excellent start!

I'd love to point you to our docs, https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#cmake for some examples

also #3951 which is not official but a great summary of what we've been following...
I fear your patches might be too large for us to accept since they will be very difficult to maintain

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 3 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py]: The package contains two components:
- a AWS SDK plugin under the `aws-cpp-sdk-cdi` subfolder which is built with cmake
- the library itself which is built with make

The library depends on the plugin and that's why I need to install it here and tell the library where to find it

File: recipes/aws-cdi-sdk/all/patches/001-Makefile.patch (hunk 5 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/patches/001-Makefile.patch:118]: the original package builds libfabric and the AWS SDK itself and here I remove that part

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 2 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:73]: see below

**Comment by dvirtz** [comment]
The way the original package is built is by downloading the AWS SDK, copying the `aws-cpp-sdk-cdi` under it and build the SDK together with the `aws-cpp-sdk-cdi` part.
To make it work with the conan AWS SDK I had to copy some files from AWS SDK to be able to build `aws-cpp-sdk-cdi` standalone. 
That's the reason for the large patches.

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 7]
```diff
@@ -16,6 +17,10 @@ class AwsCdiSdkConan(ConanFile):
     exports_sources = ["CMakeLists.txt", "patches/**"]
     generators = "cmake", "cmake_find_package"
     requires = "aws-libfabric/1.9.1amzncdi1.0", "aws-sdk-cpp/1.8.130"
+    default_options = {
+        "aws-libfabric:shared": True,
+        "aws-sdk-cpp:shared": True
+    }
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:23]: Looks like we are missing some docs... I do not know of any recipe currently that overrides defaults like this (we typically refrain from doing it in other places as well)

🟨 Can anyone confirm? i'd love to add more docs

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 8]
```diff
@@ -24,19 +29,17 @@ def _source_subfolder(self):
...
 
     def validate(self):
         if self.settings.os != "Linux":
-            raise ConanInvalidConfiguration("The aws-cdi-sdk con only be built on Linux.")
+            raise ConanInvalidConfiguration("This recipe currently only supports Linux. Feel free to contribute other platforms!")
         elif (self.settings.compiler == "gcc"
                 and tools.Version(self.settings.compiler.version) < "6.0"):
             raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
             See https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-802631744""")
+        if not self.options["aws-libfabric"].shared or self.options["aws-sdk-cpp"].shared:
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:40]: I am not sure how python handles this but maybe this is easier to read?

```suggestion
        if not self.options["aws-libfabric"].shared or not self.options["aws-sdk-cpp"].shared:
```

File: recipes/aws-cdi-sdk/all/patches/002-CMake.patch [hunk 9]
```diff
@@ -33,6 +33,8 @@ index 9ca1679..193e136 100644
 +    set(ARCHIVE_DIRECTORY "${LIBRARY_DIRECTORY}")
 +endif()
 +
++set(CPP_STANDARD "11" CACHE STRING "Flag to upgrade the C++ standard used. The default is 11. The minimum is 11.")
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/patches/002-CMake.patch:36]: Maybe set it to https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html so conan can inject the users settings?

**Comment by prince-chrismc** [comment]
🤢 https://github.com/aws/aws-cdi-sdk/blob/mainline/INSTALL_GUIDE_LINUX.md#download-aws-sdk

I see why this was so intrusive

**Review by prince-chrismc — COMMENTED** [review]
I am not familiar with this project so I apologize if my questions seem obvious.

If the Makefile calls CMake only for the AWS SDK (which is now a conan dep) https://github.com/aws/aws-cdi-sdk/blob/4a00ff961dea88e98bf7b02c1898c7c3f1dcee34/Makefile#L379-L380
deleted in the patch https://github.com/conan-io/conan-center-index/pull/7407/files/696dc90c2e0c7440d077d95c5d8ab99893a62cca..c9124a81a5b82a654042eec27872c750a7930922#diff-d886a1075a44e26aa2b7ac51f6b478b7f7e6c91b84bf1769daa772f39f12fe2bR212-R213 why does it need to be called for this project?

Can you provide some links and explain why there are to build steps for this project?

***

Instead of patching in large sections of the parent projects source code... could we download the correctly files?

https://github.com/conan-io/conan-center-index/pull/7407/files/696dc90c2e0c7440d077d95c5d8ab99893a62cca..c9124a81a5b82a654042eec27872c750a7930922#diff-b03751ea47146fed761422871ad770aebc12482dca62c210cf2b384be44aaf33R50-R51

Just as an example 
https://github.com/conan-io/conan-center-index/blob/28e0cc83ed694811c3da3971a251ef2cce3c59d9/recipes/yasm/all/conanfile.py#L41-L42

or this 

https://github.com/conan-io/conan-center-index/blob/a11644d9b1d1f41f7928122631329d21296226d4/recipes/theora/all/conanfile.py#L49

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 7 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:23]: without it all the CI builds will get `ConanInvalidConfiguration`

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 8 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:40]: good catch

**Review by dvirtz — COMMENTED** [review]
In the original package `make` is calling `cmake`: https://github.com/aws/aws-cdi-sdk/blob/a0d0cbf499e3c96ce3510810f58ae96bb6350480/Makefile#L342

I though it would be better if `conan` would call `cmake` directly so it can inject settings as you mentioned.

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 7 above)
Comment (prince-chrismc) [reply recipes/aws-cdi-sdk/all/conanfile.py:23]: Why are static not supported?

Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:23]: I tried and it doesn't compile or it crashes. I didn't invested some time in debugging it but couldn't come up with a working solution. The original package builds dynamic dependencies unconditionally: https://github.com/aws/aws-cdi-sdk/blob/a0d0cbf499e3c96ce3510810f58ae96bb6350480/Makefile#L134

File: recipes/aws-cdi-sdk/all/conandata.yml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+sources:
+  "2.2.0":
+    - url: https://github.com/aws/aws-cdi-sdk/archive/refs/tags/v2.2.0.tar.gz
+      sha256: 4ee109b7fa2f683b4b0cd03028d0946857f09f34da4e89fe69a5c297cdaeb689
+    - url: https://raw.githubusercontent.com/aws/aws-sdk-cpp/1.8.130/cmake/compiler_settings.cmake
```
Comment (SSE4) [inline recipes/aws-cdi-sdk/all/conandata.yml:5]: why does `2.2.0` downloads some sources from `1.8.130`? it looks suspicious...
are they missing from the distribution? if so, why not take them from 2.2.0 tag?

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 7 above)
Comment (SSE4) [reply recipes/aws-cdi-sdk/all/conanfile.py:23]: it's legal syntax and well supported. used in several places across CCI already, for instance:
https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/thrust/all/test_package/conanfile.py#L8

Comment (prince-chrismc) [reply recipes/aws-cdi-sdk/all/conanfile.py:23]: 🤯 Awesome thanks for confirming... I never saw it being used in CCI but now I know!

**Comment by dvirtz** [comment]
@prince-chrismc , @SSE4 
is there anything that I've missed here?

**Comment by dvirtz** [comment]
@SSE4 This PR is unblocked now that #7830 has been merged. Can you please reopen it?

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+    _cmake = None
+
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("This recipe currently only supports Linux. Feel free to contribute other platforms!")
+        elif (self.settings.compiler == "gcc"
+                and tools.Version(self.settings.compiler.version) < "6.0"):
+            raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
+            See https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-802631744""")
```
[the remark this answers is not in the document]
Comment (madebr) [reply recipes/aws-cdi-sdk/all/conanfile.py:39]: If a requirement throws a `ConanInvalidConfiguration`, cci won't build that configuration.

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        config = "debug" if self.settings.build_type == "Debug" else "release"
+        self.copy(pattern="*", dst="lib", src=os.path.join(self._source_subfolder, "build", config, "lib"))
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = self.collect_libs()
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:106]: Looking at the logs I only saw one lib being build/installed

```suggestion
        self.cpp_info.libs = ["aws-cpp-sdk-cdi"]
```

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+        if self.settings.os == "Linux":
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:107]: It hard to find but https://github.com/aws/aws-sdk-cpp/blob/bd00fe8d76e2a774c6342f659b49d4458658f4c3/cmake/utilities.cmake#L34 is the magic how they do things

The v1 generators need more love 

```suggestion
        self.cpp_info.set_property("cmake_file_name", "aws-cpp-sdk-cdi")
        self.cpp_info.set_property("cmake_target_name", "aws-cpp-sdk-cdi")

        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
        # TODO: Remove the namespace on CMake targets
        self.cpp_info.names["cmake_find_package"] = "aws-cpp-sdk-cdi"
        self.cpp_info.names["cmake_find_package_multi"] = "aws-cpp-sdk-cdi"
        if self.settings.os == "Linux":
```

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+            self.cpp_info.defines = ["_LINUX"]
+        self.cpp_info.requires = ["aws-sdk-cpp::monitoring", "aws-libfabric::aws-libfabric"]
```
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/aws-cdi-sdk/all/conanfile.py:109]: bump https://github.com/conan-io/conan-center-index/pull/7407/files#r731926015

File: recipes/aws-cdi-sdk/all/patches/002-CMake.patch [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+Date:   Mon Jul 5 09:51:24 2021 +0300
+
+    build standalone cmake
+
+diff --git a/aws-cpp-sdk-cdi/CMakeLists.txt b/aws-cpp-sdk-cdi/CMakeLists.txt
+index 9ca1679..5dcbcec 100644
+--- a/aws-cpp-sdk-cdi/CMakeLists.txt
++++ b/aws-cpp-sdk-cdi/CMakeLists.txt
+@@ -1,4 +1,39 @@
+-add_project(aws-cpp-sdk-cdi "C++ SDK for the AWS cdi service" aws-cpp-sdk-core)
```
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/aws-cdi-sdk/all/patches/002-CMake.patch:12]: Maybe the `find_package` could be called from the wrapper?

File: recipes/aws-cdi-sdk/all/test_package/CMakeLists.txt [hunk 16]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup(TARGETS)
+
+find_package(aws-cdi-sdk REQUIRED)
+
+add_executable(${PROJECT_NAME} test_package.c)
+target_link_libraries(${PROJECT_NAME} aws-cdi-sdk::aws-cdi-sdk)
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/test_package/CMakeLists.txt:10]:
```suggestion
find_package(aws-cpp-sdk-cdi REQUIRED)

add_executable(${PROJECT_NAME} test_package.c)
target_link_libraries(${PROJECT_NAME} aws-cpp-sdk-cdi::aws-cpp-sdk-cdi)
```

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 17]
```diff
@@ -0,0 +1,110 @@
+import os
+import re
+
+from conans import AutoToolsBuildEnvironment, CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.35.0"
+
+class AwsCdiSdkConan(ConanFile):
+    name = "aws-cdi-sdk"
```
Comment (prince-chrismc) [inline recipes/aws-cdi-sdk/all/conanfile.py:10]: The internal name is `aws-cpp-sdk-cdi` but this matches the repo 🤔 not sure which to match

Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:10]: this package contains two libraries: aws-cpp-sdk-cdi and cdisdk which is a C library

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:109]: `sdk-cpp::monitoring` is validated. `aws-libfabric::aws-libfabric` is the main component.

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 13 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:107]: I've split it to two components, one for the C++ SDK named `aws-cpp-sdk-cdi` and one for the C SDK named `cdisdk` like the library.

File: recipes/aws-cdi-sdk/all/test_package/CMakeLists.txt (hunk 16 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/test_package/CMakeLists.txt:10]: the package name is `aws-cdi-sdk`

File: recipes/aws-cdi-sdk/all/patches/002-CMake.patch (hunk 15 above)
[the remark this answers is not in the document]
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/patches/002-CMake.patch:12]: I ended up removing the patch entirely

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 13 above)
Comment (prince-chrismc) [reply recipes/aws-cdi-sdk/all/conanfile.py:114]: We still need to set the CMake and pkg config files names 🙏 pretty please

Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py:124]: done

**Review by SSE4 — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/aws-cdi-sdk"
+    license = "BSD-2-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    default_options = {
+        "aws-libfabric:shared": True,
+        "aws-sdk-cpp:shared": True
+    }
```
Comment (SpaceIm) [inline recipes/aws-cdi-sdk/all/conanfile.py:22]: Don't do that. Force these values in `configure()`, then check in `validate()`.

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+        
+        cppSdk.set_property("cmake_file_name", "aws-cpp-sdk-cdi")
+        cppSdk.set_property("cmake_target_name", "aws-cpp-sdk-cdi")
+        cppSdk.set_property("pkg_config_name", "aws-cpp-sdk-cdi")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        # TODO: Remove the namespace on CMake targets
+        cppSdk.names["cmake_find_package"] = "aws-cpp-sdk-cdi"
+        cppSdk.names["cmake_find_package_multi"] = "aws-cpp-sdk-cdi"
+        cppSdk.names["pkg_config"] = "aws-cpp-sdk-cdi"
```
Comment (SpaceIm) [inline recipes/aws-cdi-sdk/all/conanfile.py:119]: it's not consistent.

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+
+        cSdk.set_property("cmake_file_name", "aws-cdi-sdk")
+        cSdk.set_property("cmake_target_name", "aws-cdi-sdk")
+        cSdk.set_property("pkg_config_name", "aws-cdi-sdk")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        # TODO: Remove the namespace on CMake targets
+        cSdk.names["cmake_find_package"] = "aws-cdi-sdk"
+        cSdk.names["cmake_find_package_multi"] = "aws-cdi-sdk"
+        cSdk.names["pkg_config"] = "aws-cdi-sdk"
```
Comment (SpaceIm) [inline recipes/aws-cdi-sdk/all/conanfile.py:135]: Same question than https://github.com/conan-io/conan-center-index/pull/7407/files#r815771311

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 19 above)
Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py]: I've set both to be `AWS::aws-cpp-sdk-cdi` similar to the aws-sdk-cpp package. 
I couldn't anything existing practice upstream.

Comment (SpaceIm) [reply recipes/aws-cdi-sdk/all/conanfile.py]: If you don't know how to deduce imported targets from upstream CMakeLists (not always obvious), you inspect packaged files locally:

- comment out the deletion of cmake files during packaging
- inspect `lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-targets.cmake` in package folder and look for `add_library()`

From what I can see in build log, `cmake_file_name` is `aws-cpp-sdk-cdi`:

```
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-targets.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-targets-release.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-config.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-config-version.cmake
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+        cSdk.set_property("cmake_file_name", "aws-cdi-sdk")
+        cSdk.set_property("cmake_target_name", "AWS::aws-cdi-sdk")
+        cSdk.set_property("pkg_config_name", "aws-cdi-sdk")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        # TODO: Remove the namespace on CMake targets
+        cSdk.names["cmake_find_package"] = "aws-cdi-sdk"
+        cSdk.names["cmake_find_package_multi"] = "aws-cdi-sdk"
+        cSdk.names["pkg_config"] = "aws-cdi-sdk"
+
```
Comment (SpaceIm) [inline recipes/aws-cdi-sdk/all/conanfile.py:141]: still wrong and inconsistent

File: recipes/aws-cdi-sdk/all/conanfile.py [hunk 22]
```diff
@@ -0,0 +1,143 @@
...
+        self.options["aws-sdk-cpp"].shared = True
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("This recipe currently only supports Linux. Feel free to contribute other platforms!")
+        if not self.options["aws-libfabric"].shared or not self.options["aws-sdk-cpp"].shared:
+            raise ConanInvalidConfiguration("Cannot build with static dependencies")
+        if not getattr(self.options["aws-sdk-cpp"], "monitoring"):
+            raise ConanInvalidConfiguration("This package requires the monitoring AWS SDK")
+        tools.check_min_cppstd(self, 11)
```
Comment (SpaceIm) [inline recipes/aws-cdi-sdk/all/conanfile.py:42]:
```suggestion
        if self.settings.compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, 11)
```

File: recipes/aws-cdi-sdk/all/conanfile.py (hunk 19 above)
Comment (SpaceIm) [reply recipes/aws-cdi-sdk/all/conanfile.py]: I see. Very confusing build system: why do they mix CMakeLists to build aws-cpp-sdk-cdi  & raw Makefile to build aws-cdi-sdk?

Comment (dvirtz) [reply recipes/aws-cdi-sdk/all/conanfile.py]: I guess different people worked on each component

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**